container-search/src/main/java/com/yahoo/search/query/profile/QueryProfileProperties.java

```diff
@@ -300,13 +300,7 @@ private Map<String, String> contextWithZoneInfo(Map<String, String> context) {
         if (zoneInfo == ZoneInfo.defaultInfo()) return context;
         if (context == null || context.isEmpty()) return zoneContext;
         if (context == zoneContext) return context;
-        if (context.containsKey(ENVIRONMENT) && context.containsKey(REGION) && context.containsKey(INSTANCE)) return context;
-
-        Map<String, String> contextWithZoneInfo = new HashMap<>(context);
-        contextWithZoneInfo.putIfAbsent(ENVIRONMENT, zoneInfo.zone().environment().name());
-        contextWithZoneInfo.putIfAbsent(REGION, zoneInfo.zone().region());
-        contextWithZoneInfo.putIfAbsent(INSTANCE, zoneInfo.application().instance());
-        return Collections.unmodifiableMap(contextWithZoneInfo);
+        return new ChainedMap(zoneContext, context);
```

Comment:
Since you were using putIfAbsent, I would expect context to be primary.

Reply:
I reorder arguments and added some tests

Reply:
Yes, on reflection, overriding this in the query would be "not good".